modules/core/src/main/java/org/apache/ignite/internal/processors/security/SecurityContextHolder.java

```diff
@@ -17,44 +17,10 @@
 
 package org.apache.ignite.internal.processors.security;
 
-import org.jetbrains.annotations.Nullable;
-
 /**
- * Thread-local security context.
+ * Representation of Security Context Holder.
  */
-public class SecurityContextHolder {
-    /** Context. */
-    private static final ThreadLocal<SecurityContext> CTX = new ThreadLocal<>();
-
-    /**
-     * Get security context.
-     *
-     * @return Security context.
-     */
-    @Nullable public static SecurityContext get() {
-        return CTX.get();
-    }
-
-    /**
-     * Set security context.
-     *
-     * @param ctx Context.
-     * @return Old context.
-     */
-    public static SecurityContext push(@Nullable SecurityContext ctx) {
-        SecurityContext oldCtx = CTX.get();
-
-        CTX.set(ctx);
-
-        return oldCtx;
-    }
-
-    /**
-     * Pop security context.
-     *
-     * @param oldCtx Old context.
-     */
-    public static void pop(@Nullable SecurityContext oldCtx) {
-        CTX.set(oldCtx);
-    }
+public interface SecurityContextHolder extends AutoCloseable {
```

Comment:
Do we still need interface here?
Can we replace it with the single implementation.

Reply:
No, we can't because of this interface has two implementations.

Reply:
done
the interface was removed